bin/sozo/src/ops/register.rs

```diff
@@ -1,21 +1,71 @@
+use std::collections::HashMap;
+
 use anyhow::{Context, Result};
-use dojo_world::contracts::WorldContract;
+use dojo_world::contracts::model::ModelReader;
+use dojo_world::contracts::{WorldContract, WorldContractReader};
+use dojo_world::manifest::DeployedManifest;
 use dojo_world::metadata::Environment;
 use dojo_world::utils::TransactionWaiter;
+use scarb::core::Config;
 use starknet::accounts::Account;
+use starknet::core::types::{BlockId, BlockTag};
 
 use crate::commands::register::RegisterCommand;
 
-pub async fn execute(command: RegisterCommand, env_metadata: Option<Environment>) -> Result<()> {
+pub async fn execute(
+    command: RegisterCommand,
+    env_metadata: Option<Environment>,
+    config: &Config,
+) -> Result<()> {
     match command {
         RegisterCommand::Model { models, world, starknet, account, transaction } => {
             let world_address = world.address(env_metadata.as_ref())?;
             let provider = starknet.provider(env_metadata.as_ref())?;
 
+            let world_reader = WorldContractReader::new(world_address, &provider)
+                .with_block(BlockId::Tag(BlockTag::Pending));
+            let remote_manifest = {
+                match DeployedManifest::load_from_remote(&provider, world_address).await {
+                    Ok(manifest) => Some(manifest),
+                    Err(e) => {
+                        return Err(anyhow::anyhow!("Failed to build remote World state: {e}"));
+                    }
+                }
+            };
+
+            let manifest = remote_manifest.unwrap();
+            let registered_models_names =
+                manifest.models.into_iter().map(|m| m.name.to_string()).collect::<Vec<String>>();
```

Comment:
```suggestion
            let registered_models_names =
                manifest.models.iter().map(|m| m.name.as_str());
```
No reason to turn into `String`, the rest of the code only use `&str` of the `name`. Also no need to call `.collect()`, we can use the iterator straight away in the `for` loop.